include/boost/gil/algorithm.hpp

```diff
@@ -734,10 +734,28 @@ void default_construct_pixels(View const& view)
 
 namespace detail {
 
+enum class copy_planarity_condition
+{
+    planar_2_planar,
+    interleaved_2_planar,
+    mixed_2_interleaved
+};
+
+using planar_2_planar_type = std::integral_constant<copy_planarity_condition,
+                                                        copy_planarity_condition::planar_2_planar>;
+using interleaved_2_planar_type = std::integral_constant<copy_planarity_condition,
+                                                        copy_planarity_condition::interleaved_2_planar>;
+using mixed_2_interleaved_type = std::integral_constant<copy_planarity_condition,
+                                                        copy_planarity_condition::mixed_2_interleaved>;
+
 /// std::uninitialized_copy for pairs of planar iterators
 template <typename It1, typename It2>
 BOOST_FORCEINLINE
-void uninitialized_copy_aux(It1 first1, It1 last1, It2 first2, std::true_type)
+void uninitialized_copy_aux(It1 first1,
+                            It1 last1,
+                            It2 first2, 
+                            It2 last2,
+                            planar_2_planar_type)
```

Comment:
Did you perhaps formatted this with clang-format using our (preliminary) [.clang-format](https://github.com/boostorg/gil/tree/develop/example/clang-format) configuration?

We strive to avoid (unnecessary) whitespace fields to the left.
You could either try our `.clang-format` or manually make it more compact.

In fact, in this particular case, the prototype will fit our [100 characters line limit recommendation](https://github.com/boostorg/gil/blob/develop/CONTRIBUTING.md#guidelines):

```
void uninitialized_copy_aux(It1 first1, It1 last1, It2 first2,  It2 last2, planar_2_planar_type)
```